src/main/java/com/lpvs/util/LPVSCommentUtil.java

```diff
@@ -66,4 +74,122 @@ public static String getMatchedLinesAsLink(
         log.debug("MatchedLines: " + matchedLines);
         return matchedLines;
     }
+
+    public static String reportCommentBuilder(
+            LPVSQueue webhookConfig,
+            List<LPVSFile> scanResults,
+            List<LPVSLicenseService.Conflict<String, String>> conflicts) {
+        String commitComment = "";
+
+        if (scanResults != null && scanResults.size() != 0) {
+            commitComment = "**Detected licenses:**\n\n\n";
+            for (LPVSFile file : scanResults) {
+                commitComment += "**File:** " + file.getFilePath() + "\n";
+                commitComment +=
+                        "**License(s):** " + file.convertLicensesToString(LPVSVcs.GITHUB) + "\n";
+                commitComment +=
+                        "**Component:** "
+                                + file.getComponentName()
+                                + " ("
+                                + file.getComponentFilePath()
+                                + ")\n";
+                commitComment +=
+                        "**Matched Lines:** "
+                                + LPVSCommentUtil.getMatchedLinesAsLink(
+                                        webhookConfig, file, LPVSVcs.GITHUB)
+                                + "\n";
+                commitComment += "**Snippet Match:** " + file.getSnippetMatch() + "\n\n\n\n";
+            }
+        }
+
+        if (conflicts != null && conflicts.size() > 0) {
+            StringBuilder commitCommentBuilder = new StringBuilder();
+            commitCommentBuilder.append("**Detected license conflicts:**\n\n\n");
+            commitCommentBuilder.append("<ul>");
+            for (LPVSLicenseService.Conflict<String, String> conflict : conflicts) {
+                commitCommentBuilder.append("<li>" + conflict.l1 + " and " + conflict.l2 + "</li>");
+                LPVSDetectedLicense detectedIssue = new LPVSDetectedLicense();
+                detectedIssue.setIssue(true);
+            }
+            commitCommentBuilder.append("</ul>");
+            if (null != webhookConfig.getHubLink()) {
+                commitCommentBuilder.append("(");
+                commitCommentBuilder.append(webhookConfig.getHubLink());
+                commitCommentBuilder.append(")");
+            }
+            commitComment += commitCommentBuilder.toString();
+        }
+
+        return commitComment;
+    }
```

Comment:
I see a lot of code duplication with `commentResults()` in `LPVSGitHubService`. Can we avoid duplications here?

Reply:
Originally, comment builder was moved to Util class in order to prevent code duplication. But it resulted to huge refactoring in service class (mostly tests). I suggest to approve this change and later we may optimize repo service util logic.

Reply:
OK. Let's do refactoring later.